cbcontainers/state/components/enforcer_deployment.go

```diff
@@ -80,7 +80,15 @@ func (obj *EnforcerDeploymentK8sObject) MutateK8sObject(k8sObject client.Object,
 	deployment.Spec.Template.ObjectMeta.Labels = desiredLabels
 	deployment.Spec.Template.Spec.ServiceAccountName = commonState.EnforcerServiceAccountName
 	deployment.Spec.Template.Spec.PriorityClassName = commonState.DataPlanePriorityClassName
-	deployment.Spec.Template.Spec.ImagePullSecrets = []coreV1.LocalObjectReference{{Name: commonState.RegistrySecretName}}
+	if agentSpec.Components.Settings.CreateDefaultImagePullSecrets {
```

Comment:
all of this can be extracted to a common util code

Reply:
done.